caikit_computer_vision/modules/segmentation/image_processing.py

```diff
@@ -0,0 +1,281 @@
+    else:
+        assert isinstance(inf_coco, COCO), "inf_coco must be an instance of class COCO"
+
+    assert (
+        "categories" and "images" in inf_coco.dataset
+    ), "inf_coco should contain both 'categories' and 'images' keys"
+
+    # give ids to annotations
+    # correct the placement of the score key as in coco
+    ann_id = 1
+    for ann in detectron2_out:
+        ann["id"] = ann_id
+        ann_id += 1
+        ann["iscrowd"] = 0
+        score = ann.pop("score")
+        ann.update({"attributes": {"score": score}})
+
+    # drop annotations with score < score_threshold
+    inf_coco.dataset["annotations"] = [
+        ann for ann in detectron2_out if ann["attributes"]["score"] >= score_threshold
+    ]
+
+    # convert segmentation to polygons
+    inf_coco = convert_seg_poly(inf_coco)
+
+    # delete empty segmentations and invalid segmentations
+    inf_coco.dataset["annotations"] = [
+        ann
+        for ann in inf_coco.dataset["annotations"]
+        if ann["segmentation"]
+        if len(ann["segmentation"][0]) > 6
+    ]
+
+    log.info("Calculating bounding boxes and areas...")
+
+    for idx, ann in enumerate(inf_coco.dataset["annotations"]):
+        # we usually have multiple polygon parts produced from one prediction mask
+        multipolygon = spg.MultiPolygon(
+            [
+                spg.Polygon(list(zip(poly[0::2], poly[1::2])))
+                for poly in ann["segmentation"]
+            ]
+        )
+        ann["area"] = int(multipolygon.area)
+
+    # extract seg array from list of segs and cast to list
+    for idx, ann in enumerate(inf_coco.dataset["annotations"]):
+        seg = [i.tolist() for i in ann["segmentation"]]
+        inf_coco.dataset["annotations"][idx]["segmentation"] = seg
+
+    # save annotations
+    if save_coco:
+        if not save_coco.endswith("/"):
+            save_coco = save_coco + "/"
+        with open(save_coco + "detections_poly.json", "w") as json_file:
+            json.dump(inf_coco.dataset, json_file, sort_keys=True, indent=4)
+
+    return inf_coco
+
+
+def convert_seg_poly(inf_coco: Type[COCO]) -> Type[COCO]:
+    """converts segmentation from detectron2 format to polygons.
+
+    Args:
+      inf_coco : coco object with annotations containing detectron2 formatted segmentations
+
+    Returns:
+      inf_coco : coco dict with annotations containing coco formatted polygons
+    """
+
+    def mask_to_polygons(mask):
+        """from Detectron2 code"""
+        # cv2.RETR_CCOMP flag retrieves all the contours and arranges them to a 2-level
+        # hierarchy. External contours (boundary) of the object are placed in hierarchy-1.
+        # Internal contours (holes) are placed in hierarchy-2.
+        # cv2.CHAIN_APPROX_NONE flag gets vertices of polygons from contours.
+        mask = np.ascontiguousarray(
+            mask
+        )  # some versions of cv2 does not support incontiguous arr
+        res = cv2.findContours(
```

Comment:
Can we change this to `res, hierarchy = cv2.findContours(...)` instead of setting this to the tuple and pulling them out? I think it's a bit less confusing